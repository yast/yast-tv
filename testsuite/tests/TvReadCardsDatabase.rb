# encoding: utf-8

module Yast
  class TvReadCardsDatabaseClient < Client
    def main
      # testedfiles: Tv.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)

      textdomain "tv"
      Yast.import "Tv"

      @READ_nil = { "target" => { "yast2" => nil } }

      @READ_empty = [
        { "target" => { "yast2" => [] } },
        { "target" => { "yast2" => {} } },
        { "target" => { "yast2" => [] } }
      ]

      @READ_db = {
        "target" => {
          "yast2" => [
            {
              "cards" => [
                {
                  "module"     => "bttv",
                  "name"       => _("Unknown bttv card"),
                  "parameters" => { "card" => "0" }
                },
                {
                  "module"     => "bttv",
                  "name"       => "TView99 CPH06X",
                  "parameters" => { "card" => "38" }
                }
              ],
              "name"  => _("Other vendors")
            },
            {
              "cards" => [
                {
                  "module"     => "bttv",
                  "name"       => "ATI TV-Wonder VE",
                  "parameters" => { "card" => "64" }
                }
              ],
              "name"  => "ATI"
            }
          ]
        }
      }

      # cannot read the database file (creates err-file due to Report::Error)
      # Tv::cards_database=nil;
      #     TEST (``(Tv::ReadCardsDatabase()), [ READ_nil, $[], $[] ], nil);

      # empty database
      Tv.cards_database = nil
      TEST(lambda { Tv.ReadCardsDatabase }, [@READ_empty, {}, {}], nil)

      # doesn't read the file (card_database is not nil)
      TEST(lambda { Tv.ReadCardsDatabase }, [@READ_db, {}, {}], nil)

      Tv.cards_database = nil
      TEST(lambda { Tv.ReadCardsDatabase }, [@READ_db, {}, {}], nil)

      nil
    end
  end
end

Yast::TvReadCardsDatabaseClient.new.main
