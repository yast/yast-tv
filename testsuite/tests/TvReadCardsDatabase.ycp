{

    include "testsuite.ycp";

    map READ = $[
	"target" : $[ "size": -1, "tmpdir" : "/tmp", ]
    ];

    TESTSUITE_INIT([READ], nil);

    textdomain "tv";
    import "Tv";

    map READ_nil =
    $[ "target":
		$[ "yast2": nil ]
	];

    map READ_empty =
    $[ "target":
		$[ "yast2": [ ] ]
	];

    map READ_db =
	$[ "target": $[ "yast2":
	[
	  $[
	    "cards" : [
		  $[
			"module" : "bttv",
	        "name" : _("Unknown bttv card"),
	        "parameters" : $[
	          "card" : "0"
	        ]
	      ],
	      $[
	        "module" : "bttv",
	        "name" : "TView99 CPH06X",
	        "parameters" : $[
	          "card" : "38"
	        ]
	      ],
	    ],
	    "name" : _("Other vendors")
	  ],
	  $[
	    "cards" : [
	      $[
	        "module" : "bttv",
	        "name" : "ATI TV-Wonder VE",
	        "parameters" : $[
	          "card" : "64"
	        ]
	      ],
		],
	    "name" : "ATI"
	  ],
	]
	]];

	// cannot read the database file (creates err-file due to Report::Error)
	/*
	Tv::cards_database=nil;
    TEST (``(Tv::ReadCardsDatabase()), [ READ_nil, $[], $[] ], nil);
	*/

	// empty database
	Tv::cards_database=nil;
	TEST (``(Tv::ReadCardsDatabase()), [ READ_empty, $[], $[] ], nil);

	// doesn't read the file (card_database is not nil)
	TEST (``(Tv::ReadCardsDatabase()), [ READ_db, $[], $[] ], nil);

	Tv::cards_database=nil;
    TEST (``(Tv::ReadCardsDatabase()), [ READ_db, $[], $[] ], nil);
}
