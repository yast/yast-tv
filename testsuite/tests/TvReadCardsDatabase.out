Read	.target.yast2 "tv_cards.ycp" []
Read	.target.yast2 "tv_dvbcards.ycp" []
Return	true
Return	true
Read	.target.yast2 "tv_cards.ycp" [$["cards":[$["module":"bttv", "name":"Unknown bttv card", "parameters":$["card":"0"]], $["module":"bttv", "name":"TView99 CPH06X", "parameters":$["card":"38"]]], "name":"Other vendors"], $["cards":[$["module":"bttv", "name":"ATI TV-Wonder VE", "parameters":$["card":"64"]]], "name":"ATI"]]
Return	true
